components/script/task_queue.rs

```diff
@@ -82,7 +88,8 @@ impl<T: QueuedTaskConversion> TaskQueue<T> {
         for msg in to_be_throttled {
             // Categorize tasks per task queue.
             let (worker, category, boxed, pipeline_id, task_source) = match msg.into_queued_task() {
-                Some((worker, category, boxed, pipeline_id, task_source)) => (worker, category, boxed, pipeline_id, task_source),
+                Some(queued_task) =>
+                    queued_task,
```

Comment:
that's a nice one, there was indeed no need to unwrap the `queued_task` right inside the match...